components/service/glean/src/main/java/mozilla/components/service/glean/DatetimeMetricType.kt

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import android.support.annotation.VisibleForTesting
+import kotlinx.coroutines.launch
+import mozilla.components.service.glean.storages.DatetimesStorageEngine
+import mozilla.components.service.glean.utils.parseISOTimeString
+import mozilla.components.service.glean.utils.truncateDate
+import mozilla.components.support.base.log.logger.Logger
+import java.util.Calendar
+import java.util.Date
+
+/**
+ * This implements the developer facing API for recording datetime metrics.
+ *
+ * Instances of this class type are automatically generated by the parsers at build time,
+ * allowing developers to record values that were previously registered in the metrics.yaml file.
+ */
+data class DatetimeMetricType(
+    override val disabled: Boolean,
+    override val category: String,
+    override val lifetime: Lifetime,
+    override val name: String,
+    override val sendInPings: List<String>,
+    val timeUnit: TimeUnit = TimeUnit.Nanosecond
```

Comment:
Since `TimeUnit.Second` is the minimum resolution, should we make this default to that?

Reply:
Good point.  Given discussion elsewhere (on Slack mainly), I think the appropriate default here is "minute" (for minimal collection reasons).